### PR TITLE
Feature/adminapi

### PR DESCRIPTION
**Pull request by fillup**

### CODE REVIEW

File: cmd/speedsnitch/main.go [hunk 1]
```diff
@@ -1,55 +1,26 @@
...
+	"time"
 )
 
 var config agent.Config
+var agentStartTime time.Time
 
 func main() {
+	agentStartTime = time.Now()
+	if len(os.Args) < 2 {
+		fmt.Println("You must provide the Admin API BaseURL as the first arguement")
```
Comment (Baggerone) [inline cmd/speedsnitch/main.go]: "argument" :-)

File: cmd/speedsnitch/main.go [hunk 2]
```diff
@@ -1,55 +1,26 @@
...
-			Format      string
-			Destination string
-		}{
-			Format:      "LogTypeFile",
-			Destination: "/var/log/speed-snitch",
-		},
+	config, err := adminapi.GetConfig(os.Args[1])
+	if err != nil {
+		fmt.Println("Unable to fetch config from admin API:", err)
+		os.Exit(1)
```
Comment (Baggerone) [inline cmd/speedsnitch/main.go]: At some point, will we want to avoid exiting for errors and handle them some other way (e.g. logging to logentries??)

File: lib/adminapi/adminapi.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		return resp, fmt.Errorf(
+			"API returned an error. \n\tMethod: %s, \n\tURL: %s, \n\tCode: %v, \n\tStatus: %s \n\tBody: %s",
+			method, url, resp.StatusCode, resp.Status, postData)
+	}
+
+	return resp, nil
+}
+
+// SayHello makes a POST call to /hello with id, version, and update
+func SayHello(config agent.Config, agentStartTime time.Time) (bool, error) {
```
Comment (Baggerone) [inline lib/adminapi/adminapi.go]: I'm thinking the bool return value isn't needed.
Either error is nil (matching your `return true, nil`) or it's not (matching your `return false, Error`).

You never have `return false, nil` or `return true, Error`

File: lib/adminapi/adminapi.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	if resp.StatusCode != 204 {
+		return false, fmt.Errorf("call to /hello did not return 204, got: %v", resp.StatusCode)
+	}
+
+	return true, nil
+}
+
+// GetConfig fetches config from
```
Comment (Baggerone) [inline lib/adminapi/adminapi.go]: "from ... ?"

File: lib/adminapi/adminapi_test.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		t.Fail()
+	}
+}
+
+func TestGetConfig(t *testing.T) {
+	mux := http.NewServeMux()
+	server := httptest.NewServer(mux)
+
+	respBody := `{
+  "BaseURL": "https://www.sil.org",
```
Comment (Baggerone) [inline lib/adminapi/adminapi_test.go]: Should these match the case of the json annotations on the structs?

File: lib/adminapi/adminapi_test.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+		fmt.Fprintf(w, respBody)
+	})
+
+	config, err := GetConfig(server.URL)
+	if err != nil {
+		t.Errorf(err.Error())
+		t.Fail()
+	}
+
+	expected := agent.Config{
```
Comment (Baggerone) [inline lib/adminapi/adminapi_test.go]: This kind of looks like you were thinking of testing that one of the other attributes of `expected` was empty.  Did you want to add that check?

File: cmd/speedsnitch/main.go (hunk 2 above)
Comment (fillup) [reply cmd/speedsnitch/main.go]: well this is before we have any config, so I'm not sure how we can report it, but I guess we'll have the base url, so if we send logs to the api we could try, but if there is an error getting the config what are the chances logging would work?

so yeah, maybe at some point we'll refactor this bit

File: lib/adminapi/adminapi.go (hunk 3 above)
Comment (fillup) [reply lib/adminapi/adminapi.go]: yeah good point, I'll change that over

File: lib/adminapi/adminapi_test.go (hunk 6 above)
Comment (fillup) [reply lib/adminapi/adminapi_test.go]: yeah, added another test for an empty Version

File: lib/adminapi/adminapi_test.go (hunk 5 above)
Comment (fillup) [reply lib/adminapi/adminapi_test.go]: yes
